### PR TITLE
Fix dashboard greetings that show 'good morning' after noon

### LINKED ISSUE #24938: Dashboard greetings shows goodmorning while it is after 12:00 (24h)
### Steps to reproduce
1. Visit Nextcloud Dashboard, after 12:00 (24H) https://<fqdn>/apps/dashboard/#/
2.
3.

### Expected behaviour
The greeting should be Good Afternoon.

### Actual behaviour
The greeting shows Good morning

See https://help.nextcloud.com/t/dashboard-time-incorrect/101461 as well.

### Server configuration

**Operating system:**  openSUSE_15.2

**Web server:** Apache

**Database:** SQLite

**PHP version:** 7.4.6

**Nextcloud version:** (see Nextcloud admin page) 20.0.4

**Updated from an older Nextcloud/ownCloud or fresh install:** older

**Where did you install Nextcloud from:** Nextcloud

### Client configuration
**Browser:** Chromium 

**Operating system:** windows / linux

### PR DESCRIPTION
**Pull request by nina-py**
Updated dashboard greetings to show "Good afternoon" after 12:00 instead of 13:00.

Updated time periods for greetings:
- Morning: 5:00 to 11.59
- Afternoon: 12:00 to 17.59
- Evening: 18:00 to 21:59
- Night: 22:00 to 4.59

Updated night-time greeting to "Hello" as "Good night" is a kind of goodbye.

Had a look at the history of that component and noted how it evolved over time. Decided to do a bit of a refactor to remove repetitive code; some code duplication is inevitable though as translation strings need to remain intact.

Closes #24938.

Signed-off-by: Nina Pypchenko <22447785+nina-py@users.noreply.github.com>

### CODE REVIEW

**Review by juliusknorr — APPROVED** [review]

**Review by julien-nc — APPROVED** [review]
:+1:

**Comment by nina-py** [comment]
Thank you for reviewing and merging this so quickly!

**Comment by graue70** [comment]
I'm pretty sure https://github.com/nextcloud/server/blob/master/apps/dashboard/src/App.vue#L169 should be a logical OR instead of AND.

File: apps/dashboard/src/App.vue [hunk 1]
```diff
@@ -163,18 +163,43 @@ export default {
...
-			const shouldShowName = this.displayName && this.uid !== this.displayName
 
-			if (time > 18) {
-				return { text: shouldShowName ? t('dashboard', 'Good evening, {name}', { name: this.displayName }) : t('dashboard', 'Good evening') }
-			}
-			if (time > 12) {
-				return { text: shouldShowName ? t('dashboard', 'Good afternoon, {name}', { name: this.displayName }) : t('dashboard', 'Good afternoon') }
+			// Determine part of the day
+			let partOfDay
+			if (time >= 22 && time < 5) {
```
Comment (graue70) [inline apps/dashboard/src/App.vue:169]: This should be a logical OR instead of AND.

Comment (eneiluj) [reply apps/dashboard/src/App.vue:169]: Thanks, see #25110 .

Comment (nina-py) [reply apps/dashboard/src/App.vue:169]: Thanks for picking this up @graue70!

**Comment by jancborchardt** [comment]
Thanks a lot for the fix @nina-py! If you would like to contribute more, we also have some community chat channels on our Nextcloud Talk instance, including specific channels e.g. for the Dashboard app and others. Let me know if you are interested and I can add you there. :)

**Comment by nina-py** [comment]
Hi @jancborchardt, my pleasure! Yes, I'd like to be added to the community chat - I've participated there as a guest. And thanks for the invite to the Nextcloud org - I'm definitely taking you up on that :).

**Pull request merged**